infrastructure/router/router.go

```diff
@@ -0,0 +1,35 @@
+package router
+
+import (
+	"log"
+
+	"github.com/wmaldonadoc/academy-go-q42021/config"
+	"github.com/wmaldonadoc/academy-go-q42021/interface/controller"
+
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+)
+
+// NewRouter - Setup & returns the API routes.
+func NewRouter(controller controller.AppController) {
+	port := config.GetEnvVariable("PORT")
+	router := gin.Default()
+	mainGroup := router.Group("/api/v1")
+	{
+		pokemonRoutes := mainGroup.Group("/pokemons")
+		{
+			pokemonRoutes.GET("/:id", func(context *gin.Context) { controller.Pokemon.GetByID(context) })
+			pokemonRoutes.GET("/name/:name", func(context *gin.Context) { controller.Pokemon.GetByName(context) })
```

Comment:
Why don't add `pokemons/id/:id` route? According to REST best practices, I should get an idea of what resource(s) I'm trying to access.

if I'm using a standard for `name`, let's use it also for `id`. Also, what do you think of adding a `GET /pokemons` endpoint if I want to get them all?


```suggestion
			pokemonRoutes.GET("pokemons/id/:id", func(context *gin.Context) { controller.Pokemon.GetByID(context) })
			pokemonRoutes.GET("pokemons/name/:name", func(context *gin.Context) { controller.Pokemon.GetByName(context) })
```